File: B/questao_2.py
# importando os dados
import matplotlib.pyplot as plt
import pandas as pd

dados = pd.read_csv("./B/questao_2.csv")
variaveis = ["Estado_civil", "Regiao_de_procedencia", "N_de_filhos", "idade"]


def prepare_plot(dados, variaveis: list[str]):
    for idx, var in enumerate(variaveis):
        plt.subplot(221 + idx)
        if dados[var].dtype == "object":
            counts = dados[var].str.strip().value_counts()
        elif pd.api.types.is_numeric_dtype(dados[var]):
            counts = dados[var].astype(int).value_counts()
        plt.bar(counts.index, counts.values)


prepare_plot(dados, variaveis)

plt.savefig("questao_2_py.png", dpi=100, bbox_inches="tight", format="png")

File: B/questao_2.r
install.packages("patchwork")
library(patchwork)
install.packages("ggplot2")
library(ggplot2)

dados <- read.csv("./questao_2.csv")

p1 <-ggplot(dados, aes(x=Estado_civil)) + geom_bar()
p2 <- ggplot(dados, aes(x=Regiao_de_procedencia)) + geom_bar()
p3 <- ggplot(dados, aes(x=N_de_filhos)) + geom_bar()
p4 <- ggplot(dados, aes(x=idade)) + geom_bar()

(p1 | p2) / (p3 | p4)


